mbbsd/register.c

```diff
@@ -1743,15 +1745,29 @@ static int notify_email_change(const char *userid, const char *email)
 //     email.
 //     0: not skip checking (to check whether they are the same.)
 //     1: skip checking (not to check whether they are the same.)
+//   two_factor_authentication:
+//     whether to do 2FA.
+//     0: don't do 2FA.
+//     1. do 2FA.
 //
 // Return:
 //   int: 0: ok -1: err
 int
-change_contact_email(bool skip_same_email_check)
+change_contact_email(bool skip_same_email_check, bool two_factor_authentication)
 {
     char email[EMAILSZ] = {};
     memcpy(email, cuser.email, sizeof(email));
```

Comment:
I think we should force a reload from disk, i.e. call `pwcuReload()`, before this, in case another login instance changed the email. This should be a new precursor commit.